packages/nuxt/src/core/nuxt.ts

```diff
@@ -122,10 +122,28 @@ async function initNuxt (nuxt: Nuxt) {
 
   // Init user modules
   await nuxt.callHook('modules:before')
-  const modulesToInstall = [
-    ...nuxt.options.modules,
-    ...nuxt.options._modules
-  ]
+  const modulesToInstall = []
+
+  const watchedPaths = new Set<string>()
+
+  // Automatically register user modules
+  for (const config of nuxt.options._layers.map(layer => layer.config).reverse()) {
+    const userModules = await resolveFiles(config.srcDir, [
```

Comment:
refactor We could rename this to `localModules` to avoid confusing with user specified modules from config

```suggestion
    const localModules = await resolveFiles(config.srcDir, [
```

Reply:
renamed to `layerModules` 👍